yt/data_objects/data_containers.py

```diff
@@ -662,6 +663,182 @@ def to_glue(self, fields, label="yt", data_collection=None):
                 pass
         else:
             data_collection.append(gdata)
+    
+    def create_firefly_object(
+        self,
+        path_to_firefly,
+        fields_to_include= [],
+        fields_units = [],
```

Comment:
It's almost never correct to use a mutable as the default value of a keyword argument. Use `None` instead and initialize these to an empty list in the function definition. See e.g. [this blog post](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments) which explains exactly why this is a bad idea.

Reply:
i am bad about this, thanks for pointing it out :]